src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentTagHelperDescriptorProvider.cs

```diff
@@ -111,38 +126,103 @@ private TagHelperDescriptor CreateDescriptor(BlazorSymbols symbols, INamedTypeSy
                     continue;
                 }
 
-                builder.BindAttribute(pb =>
+                CreateProperty(builder, property.property, property.kind);
+            }
+
+            var descriptor = builder.Build();
+
+            return descriptor;
+        }
+
+        private void CreateProperty(TagHelperDescriptorBuilder builder, IPropertySymbol property, PropertyKind kind)
+        {
+            builder.BindAttribute(pb =>
+            {
+                pb.Name = property.Name;
+                pb.TypeName = property.Type.ToDisplayString(FullNameTypeDisplayFormat);
+                pb.SetPropertyName(property.Name);
+
+                if (kind == PropertyKind.Enum)
                 {
-                    pb.Name = property.property.Name;
-                    pb.TypeName = property.property.Type.ToDisplayString(FullNameTypeDisplayFormat);
-                    pb.SetPropertyName(property.property.Name);
+                    pb.IsEnum = true;
+                }
 
-                    if (property.kind == PropertyKind.Enum)
-                    {
-                        pb.IsEnum = true;
-                    }
+                if (kind == PropertyKind.ChildContent)
+                {
+                    pb.Metadata.Add(BlazorMetadata.Component.ChildContentKey, bool.TrueString);
+                }
 
-                    if (property.kind == PropertyKind.ChildContent)
-                    {
-                        pb.Metadata.Add(BlazorMetadata.Component.ChildContentKey, bool.TrueString);
-                    }
+                if (kind == PropertyKind.Delegate)
+                {
+                    pb.Metadata.Add(BlazorMetadata.Component.DelegateSignatureKey, bool.TrueString);
+                }
+
+                if (HasTypeParameter(property.Type))
+                {
+                    pb.Metadata.Add(BlazorMetadata.Component.GenericTypedKey, bool.TrueString);
+                }
+
+                var xml = property.GetDocumentationCommentXml();
+                if (!string.IsNullOrEmpty(xml))
+                {
+                    pb.Documentation = xml;
+                }
+            });
+
+            bool HasTypeParameter(ITypeSymbol type)
+            {
+                if (type is ITypeParameterSymbol)
+                {
+                    return true;
+                }
 
-                    if (property.kind == PropertyKind.Delegate)
+                // We need to check for cases like:
+                // [Parameter] List<T> MyProperty { get; set; }
+                // AND
+                // [Parameter] List<string> MyProperty { get; set; }
```

Comment:
Not necessarily a problem, but I didn't spot any unit test coverage for the pre-closed generic types (like your `List<string>` example here) or a nested generic type. Though maybe I'm missing the subtleties of what's different about those cases. Do you feel L191-L207 is adequately covered?

Reply:
I'll add a test for make it explicit. Pre-closed generics are already including in our built-in components. This code actually took some doing to get right beause their lingo is a little confusing.